### PR TITLE
Update samples

### PR DESCRIPTION
**Pull request by dashpole**
Updates the beyla golden signals sample, and uses the `disable_informers` and `drop_external` options.  This reduces memory usage by half, and reduces CPU usage by > 90%.  It also means we can get rid of our filter in the otel collector config. In a follow-up, we could replace the collector in this sample with the k8s-otlp-ingest one so we don't need to maintain it anymore here.

The routes `unmatched: wildcard` preserves existing behavior, which is to disable the route heuristics.  It did seem to work well when I tried it with the OTel demo, but that can be removed separately.

There were a few typos in some samples, which i've fixed.

The beyla service graph sample needed the metric name "calls" updated to "traces.span.metrics.calls" because of an update to the collector component.

### CODE REVIEW

File: recipes/beyla-golden-signals/gke/beyla-daemonset.yaml [hunk 1]
```diff
@@ -122,5 +122,9 @@ data:
     attributes:
       kubernetes:
         enable: true
+        drop_external: true
```
Comment (aabmass) [inline recipes/beyla-golden-signals/gke/beyla-daemonset.yaml:125]: would be good to add a comment explaining what the options are for

File: recipes/beyla-service-graph/beyla-daemonset.yaml [hunk 2]
```diff
@@ -121,10 +121,14 @@ data:
...
       endpoint: http://otel-collector:4317
       interval: 30s
-    # Kubernetes decoration must be enabled to use k8s_pod_name service discovery
-    # https://github.com/grafana/beyla/issues/673
     attributes:
+      instance_id:
+        dns: false
       kubernetes:
         enable: true
+        drop_external: true
```
Comment (aabmass) [inline recipes/beyla-service-graph/beyla-daemonset.yaml:129]: same comment

**Review by aabmass — APPROVED** [review]

File: recipes/beyla-golden-signals/gke/beyla-daemonset.yaml (hunk 1 above)
Comment (dashpole) [reply recipes/beyla-golden-signals/gke/beyla-daemonset.yaml:126]: done

File: recipes/beyla-service-graph/beyla-daemonset.yaml (hunk 2 above)
Comment (dashpole) [reply recipes/beyla-service-graph/beyla-daemonset.yaml:130]: done

**Pull request merged**